src/USER-LD/pair_localdensity.cpp

```diff
@@ -0,0 +1,883 @@
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing authors:
+   Tanmoy Sanyal, M.Scott Shell, UC Santa Barbara
+   David Rosenberger, TU Darmstadt
+------------------------------------------------------------------------- */
+
+#include "pair_localdensity.h"
+#include <mpi.h>
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+#include <string>
+#include "atom.h"
+#include "force.h"
+#include "comm.h"
+#include "neighbor.h"
+#include "neigh_list.h"
+#include "neigh_request.h"
+#include "memory.h"
+#include "error.h"
+#include "domain.h"
+
+using namespace LAMMPS_NS;
+
+static const char cite_user_ld_package[] =
+  "USER-LD package:\n\n"
+  "@Article{sanyal_2018,\n"
+  " author = {T. Sanyal, M. S. Shell},\n"
+  " title = {Transferable coarse-grained models of liquid-liquid equilibrium using local density potentials optimized with the relative entropy},\n"
+  " journal = {J.~Phys.~Chem. B},\n"
+  " year =    2018,\n"
+  " volume =  122(21),\n"
+  " pages =   {5678-5693}\n"
+  "}\n\n";
+
+#define MAXLINE 1024
+
+/* ---------------------------------------------------------------------- */
+
+PairLOCALDENSITY::PairLOCALDENSITY(LAMMPS *lmp) : Pair(lmp)
+{
+  restartinfo = 0;
+  one_coeff = 1; 	
+  single_enable = 1;
+  
+  // stuff read from tabulated file
+  nLD = 0;
+  nrho = 0;
+  rho_min = NULL;
+  rho_max = NULL;
+  a = NULL;
+  b = NULL;
+  c0 = NULL;
+  c2 = NULL;
+  c4 = NULL;
+  c6 = NULL;
+  uppercut = NULL;
+  lowercut = NULL;
+  uppercutsq = NULL;
+  lowercutsq = NULL;
+  frho = NULL;
+  rho = NULL;
+  
+  // splined arrays
+  frho_spline = NULL;
+  
+  // per-atom arrays
+  nmax = 0;
+  fp = NULL;
+  localrho = NULL;  
+
+  // set comm size needed by this pair
+  comm_forward = 1;
+  comm_reverse = 1;
+}
+
+/* ----------------------------------------------------------------------
+   check if allocated, since class can be destructed when incomplete
+------------------------------------------------------------------------- */
+
+PairLOCALDENSITY::~PairLOCALDENSITY()
+{
+
+  memory->destroy(localrho);
+  memory->destroy(fp);
+
+  if (allocated) {
+    memory->destroy(setflag);
+    memory->destroy(cutsq);
+  }
+
+  memory->destroy(frho_spline);
+  
+  memory->destroy(rho_min);  
+  memory->destroy(rho_max);
+  memory->destroy(delta_rho);	
+  memory->destroy(c0);
+  memory->destroy(c2);
+  memory->destroy(c4);
+  memory->destroy(c6);
+  memory->destroy(uppercut);
+  memory->destroy(lowercut);
+  memory->destroy(uppercutsq);
+  memory->destroy(lowercutsq);
+  memory->destroy(frho);
+  memory->destroy(rho);
+
+  delete [] a;
```

Comment:
`a` and `b` are allocated via `memory->create()` and not `new`, thus they should be freed using `memory->destroy()`. The memory class uses `malloc()/free()` internally which should not be mixed with `new/delete`. Some compilers will throw errors, if this mix happens.